components/prism-tt2.js

```diff
@@ -0,0 +1,78 @@
+(function(Prism) {
+
+	var tt2_pattern = /\[%[\s\S]+?%\]|^%%*?/;
+
+	Prism.languages.tt2 = Prism.languages.extend('markup', {
+		'tt2': {
+			pattern: tt2_pattern,
+			inside: {
+				'comment': {
+					pattern: /(^|[^\\$])#.*/,
+					lookbehind: true
+				},
+				'delimiter': {
+					pattern: /^(?:\[%|%%)-?|-?%]$/i,
+					alias: 'punctuation'
+				},
+				'string': /(["'])(?:\\.|(?!\1)[^\\\r\n])*\1/,
+				'number': /\b0x[\dA-Fa-f]+\b|(?:\b\d+\.?\d*|\B\.\d+)(?:[Ee][+-]?\d+)?/,
+				'keyword': /\b(?:GET|CALL|SET|DEFAULT|INSERT|INCLUDE|PROCESS|WRAPPER|BLOCK|IF|UNLESS|ELSIF|ELSE|SWITCH|CASE|FOREACH|WHILE|FILTER|USE|MACRO|RAWPERL|PERL|TRY|THROW|CATCH|FINAL|NEXT|LAST|RETURN|STOP|CLEAR|META|TAGS|DEBUG)\b/,
+				'operator': /\b(?:=>|==|!=|<|<=|>|>=|&&\|\|!|and|or|not)\b/,
+				'punctuation': /[[\]{},]/,
+				'variable': /[a-zA-Z][\.\/\$a-zA-Z0-9]*/
+			}
+		}
+	});
+
+	// Tokenize all inline TT2 expressions that are wrapped in [% %]
+	// This allows for easy TT2 + markup highlighting
+	Prism.hooks.add('before-highlight', function(env) {
```

Comment:
As previously expressed, this whole code can now be drastically simplified by using the functions available in the Markup-templating component. Plus it will make the component fully compatible with NodeJS.